Source/WTF/wtf/glib/WTFGType.h

```diff
@@ -43,6 +43,18 @@ static void destroy##structName(structName* data) \
 #define WEBKIT_DEFINE_ABSTRACT_TYPE(TypeName, type_name, TYPE_PARENT) _WEBKIT_DEFINE_TYPE_EXTENDED(TypeName, type_name, TYPE_PARENT, G_TYPE_FLAG_ABSTRACT, { })
 #define WEBKIT_DEFINE_TYPE_WITH_CODE(TypeName, type_name, TYPE_PARENT, Code) _WEBKIT_DEFINE_TYPE_EXTENDED_BEGIN(TypeName, type_name, TYPE_PARENT, 0) {Code;} _WEBKIT_DEFINE_TYPE_EXTENDED_END()
 
+// Only the 2022 API uses final types for now. If the old API ever gains
+// a final type, move the corresponding macro above out of the #if block.
+#if ENABLE(2022_GLIB_API)
+#define WEBKIT_DEFINE_FINAL_TYPE(TypeName, type_name, TYPE_PARENT) _WEBKIT_DEFINE_TYPE_EXTENDED(TypeName, type_name, TYPE_PARENT, G_TYPE_FLAG_FINAL, { })
+#define WEBKIT_DEFINE_FINAL_TYPE_WITH_CODE(TypeName, type_name, TYPE_PARENT, Code) _WEBKIT_DEFINE_TYPE_EXTENDED_BEGIN(TypeName, type_name, TYPE_PARENT, G_TYPE_FLAG_FINAL) { Code; } _WEBKIT_DEFINE_TYPE_EXTENDED_END()
+#define WEBKIT_DEFINE_FINAL_TYPE_IN_2022_API WEBKIT_DEFINE_FINAL_TYPE
+#define WEBKIT_DEFINE_FINAL_TYPE_WITH_CODE_IN_2022_API WEBKIT_DEFINE_FINAL_TYPE_WITH_CODE
+#else
+#define WEBKIT_DEFINE_FINAL_TYPE_IN_2022_API WEBKIT_DEFINE_TYPE
+#define WEBKIT_DEFINE_FINAL_TYPE_WITH_CODE_IN_2022_API WEBKIT_DEFINE_TYPE_WITH_CODE
```

Comment:
This is fine, but I'm not sure it's really much benefit? I would just provide two different definitions of `WEBKIT_DEFINE_FINAL_TYPE`, where it uses `G_TYPE_FLAG_FINAL` if it's the new API and just doesn't for the old API. That will avoid adding the `_IN_2022_API` cruft to all the source files.

Would that conflict with what you've got planned next?

Reply:
I thought of that first, but then it seemed to me quite obscure that a `WEBKIT_DEFINE_FINAL_TYPE` macro won't create a final type when `ENABLE(2022_GLIB_API)` is off&mdash;it feels very wrong and _will_ be confusing. So I did the extra indirection with the `_IN_2022_API` suffix for clarity.